modules/platforms/dotnet/Apache.Ignite.Core/Impl/Binary/BinaryHashCodeUtils.cs

```diff
@@ -45,11 +45,33 @@ public static unsafe int GetHashCode<T>(T val, Marshaller marsh, IDictionary<int
             if (type == typeof(long))
                 return GetLongHashCode(TypeCaster<long>.Cast(val));
 
+            if (type == typeof(string))
+                return BinaryUtils.GetStringHashCode((string) (object) val);
+
+            if (type == typeof(Guid))
+                return GetGuidHashCode(TypeCaster<Guid>.Cast(val));
+
+            if (type == typeof(uint))
+            {
+                var val0 = TypeCaster<uint>.Cast(val);
+                return *(int*) &val0;
+            }
+
+            if (type == typeof(ulong))
+            {
+                var val0 = TypeCaster<ulong>.Cast(val);
+                return GetLongHashCode(*(long*) &val0);
+            }
+
```

Comment:
What about all these types? They was not supported also?

Reply:
They were supported - check the diff. I've reordered the checks, so that more likely use cases come early.

For example, I can imaging that `long`, `Guid` and `string` are among the most popular key type choices, while `double` would be very weird.